### PR TITLE
Sets lists implementation

### PR DESCRIPTION
**Pull request by rvyuha**
Creating temporary PR to get feedback on display format for lists.

### CODE REVIEW

**Comment by rvyuha** [comment]
Something of note:
Identical sets and lists do not have identical displays.
Using the side bar navigation takes you to the first instance of the anchor.

**Review by DougManuel — CHANGES_REQUESTED** [review]
1) There is an error in the output for the first set, 

```
## Aggregations { #aggregation } [(aggregation)](file:///Users/dmanuel/github/PHES-ODM-Doc/docs/parts.html#aggregation). Statistical measures used to report a measure. Each measure reported as a number should be reported with an aggregation.
```

1) For all heading # 3: (i.e. `Boolean aggregation set`), add the following.
- On the next line, print the (partID). label. description. instruction.
- Then: status, first released, last updated.
The partID should have the link to the partID.

1) For `Category Sets`. 
- Change all the Heading # 3 to heading #2. (meaning move heading three out of the nest within `Category Sets`. 
- Remove heading #3 (category sets).

We can merge to main after that update.

**Comment by DougManuel** [comment]
One additional feature that is highish priority is to update all links **to** the sets page. 
- Any reference to a partID attribute that doesn't begin with 'na' and ends in 'catSet' or 'Set' should link to set.html. The link is currently parts.html.
- This applies to the following attributes:
     - Compartment Set
     - Unit Set ID
     - Aggregation Scale ID
     - Quality Set ID
     - Category Set ID
     - Unit Set ID

**Comment by DougManuel** [comment]
The PR looks good. The first part is rendering without a problem.

The version yesterday had a collapsable navigation bar on the right side, but it doesn't today. I like the collapsable nav. 

I pushed version rc.3.10. Why don't you update with that version and we'll see if the duplicates get removed. Then we should be GTG.

File: qmd/sets.qmd [hunk 1]
```diff
@@ -22,119 +22,351 @@ source("R/qmd-utils.R")
...
+    set_display <- glue::glue(
+      '{set_display}
+<a href=\"/parts.html#{set_ID}\">({set_ID})</a>{set_ID_part_description} \\
+{if(set_ID_part_instruction %!=na% constants$dictionary_missing_value_replacement) set_ID_part_instruction else ""} </br>
+Status: {set_ID_part_status}. First Released: {set_ID_part_first_released}. \\
+Last updated: {set_ID_part_last_updated}.'
+    )
+    
     # Create list to hold a values for html list
     list_values <- list()
```
Comment (yulric) [inline qmd/sets.qmd:149]: Should be named more specific, right now I don't know what list values its supposed to hold. For example, `set_cat_list_values`.

File: qmd/sets.qmd [hunk 2]
```diff
@@ -22,119 +22,351 @@ source("R/qmd-utils.R")
...
+    verify_input(
+      sub_category_first_released,
+      glue::glue('Part first released is missing for {sub_category_ID}')
+    )
+    verify_input(sub_category_last_updated,
+                 glue::glue('Part last updates is missing for {sub_category_ID}'))
+    
+    sub_categories_list[[sub_category_ID]] <-
+      glue::glue(
+        '**{sub_category_label}** <a href=\"/parts.html#{sub_category_ID}\">({sub_category_ID})</a>. {sub_category_description} </br>Status: {sub_category_status}. First released: {sub_category_first_released}. Last updated: {sub_category_last_updated}.'
```
Comment (yulric) [inline qmd/sets.qmd:356]: This is the same template string as on lines https://github.com/Big-Life-Lab/PHES-ODM-Doc/blob/sets-lists-implementation/qmd/sets.qmd#L195 and https://github.com/Big-Life-Lab/PHES-ODM-Doc/blob/sets-lists-implementation/qmd/sets.qmd#L280. Probably should be moved into a function.

File: qmd/sets.qmd [hunk 3]
```diff
@@ -22,119 +22,351 @@ source("R/qmd-utils.R")
...
+    set_type_instruction,
+    glue::glue('Category instructions are missing for {category_set_ID}')
+  )
+  category_description <-
+    category_info[[parts_sheet_column_names$part_description_column_name]]
+  verify_input(set_type_description,
+               glue::glue('Category description is missing for {category_set_ID}'))
+  
+  category_display <- glue::glue(
+    '## {category_label} {{ #{category_set_ID} }}
```
Comment (yulric) [inline qmd/sets.qmd:313]: This template string is the same as on lines https://github.com/Big-Life-Lab/PHES-ODM-Doc/blob/sets-lists-implementation/qmd/sets.qmd#L233-L236 and https://github.com/Big-Life-Lab/PHES-ODM-Doc/blob/sets-lists-implementation/qmd/sets.qmd#L89-L91 and should be moved into a function.

**Review by yulric — CHANGES_REQUESTED** [review]

**Comment by yulric** [comment]
I also noticed that the content structure for the different set types (list, catSet, setType) are the same. The main difference is what the grouping is (list, catSet, setID) and what the categories are in that grouping. I believe the code can be significantly reduced and made less confusing if we make one function to render each grouping and use it everywhere. Let me know if this does not make sense and we can discuss it.

**Comment by DougManuel** [comment]
Yes, those lists are the same, essentially. I am talking this morning to @mathew-thomson about a modest clean-up renaming of these parts (spurred by this work). It shouldn't change the structure and location of the parts -- at least for version 2. I'll put a Discourse post in this discussion.

**Pull request merged**